src/Controls/tests/TestCases.Shared.Tests/Tests/CollectionView/CollectionViewUITests.HeaderAndFooter.cs

```diff
@@ -0,0 +1,94 @@
+using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests
+{
+#if IOS
+    public class CollectionViewHeaderAndFooterTests : CollectionViewUITests
+    {
+
+        protected override bool ResetAfterEachTest => true;
+
+        public CollectionViewHeaderAndFooterTests(TestDevice device)
+            : base(device)
+        {
+        }
+
+        [Test]
+        [Category(UITestCategories.CollectionView)]
+        public void HeaderFooterStringWorks()
+        {
+            // Navigate to the selection galleries
+            VisitInitialGallery("Header Footer");
+
+            // Navigate to the specific sample inside selection galleries
+            VisitSubGallery("Header/Footer (String)");
+
+            App.WaitForElement("Just a string as a header");
+            App.WaitForElement("This footer is also a string");
+        }
+
+        [Test]
+        [Category(UITestCategories.CollectionView)]
+        public void HeaderFooterViewWorks()
+        {
+            // Navigate to the selection galleries
+            VisitInitialGallery("Header Footer");
+
+            // Navigate to the specific sample inside selection galleries
+            VisitSubGallery("Header/Footer (Forms View)");
+
+            App.WaitForElement("This Is A Header");
+            App.WaitForElement("This Is A Footer");
+        }
+
+        [Test]
+        [Category(UITestCategories.CollectionView)]
+        public void HeaderFooterTemplateWorks()
+        {
+            // Navigate to the selection galleries
+            VisitInitialGallery("Header Footer");
+
+            // Navigate to the specific sample inside selection galleries
+            VisitSubGallery("Header/Footer (Template)");
+
+            VerifyScreenshot();
+        }
+
+        [Test]
+        [Category(UITestCategories.CollectionView)]
+        public void HeaderFooterGridWorks()
+        {
+            // Navigate to the selection galleries
+            VisitInitialGallery("Header Footer");
+
+            // Navigate to the specific sample inside selection galleries
+            VisitSubGallery("Header/Footer (Grid)");
+
+            App.WaitForElement("This Is A Header");
```

Comment:
The tests are failing waiting for headers/footers. "This Is A header" is not an AutomationId right? We are adding more possibilities to WaitForElement to allow use queries and search by text content etc.  https://github.com/dotnet/maui/pull/25177

For now, could do something like:
```
var header = app.Driver.FindElement(OpenQA.Selenium.By.XPath("//*[@text='" + "This Is A Header" + "']"));
ClassicAssert.IsNotNull(header); // The Header exists, is in the hierarchy and we can find it.
```